src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -291,6 +291,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let offset_ptr = ptr.ptr_wrapping_signed_offset(offset_bytes, self);
                 self.write_scalar(offset_ptr, dest)?;
             }
+            sym::ptr_guaranteed_eq | sym::ptr_guaranteed_ne => {
```

Comment:
I just realized that this implementation will also be used by Miri, which is a bug. Can we somehow make it be used only during CTCE?

Reply:
doesn't miri fall back to this only if it has no own impl? So we can just give miri an impl that behaves differently.

Reply:
Miri prefers the CTFE impls, to make sure we test those properly.

I am working on a PR that adds these to Miri in a way that it prefers its own impls.

Reply:
Here it is: https://github.com/rust-lang/miri/pull/1459